todo-list/script.js

```diff
@@ -1,150 +1,314 @@
-                    </div>
-                </label>
-            </li>
-        `;
-        wrapperList.innerHTML = todoString;
-    });
-    
-    saveToLocalStorageState();
+class Task {
+    constructor(label, id) {
+        this.label = label;
+        this.isCompeleted = false;
+        this.taskId = id;
+    }
 }
 
-function addTodo(event) {
-    event.preventDefault();
-    const label = this.value.trim();
-    let checked = false;
-    if(this.value != '' && !todos.find(todo => todo == this.value)) {
-        todos = [
-            ...todos,
-            {
-                label: this.value,
-                checked
+class TodoList {
+    
+    constructor(elementInitApp, preset = {}) {
+        this.titleOfApp = preset.title || 'Todo List App';
+        this.idOfApp = preset.id || '';
+        this.elementInitApp = elementInitApp;
+        this.state = [];
+        this.arrayOfLiElementsToaster = [];
+        this.isAlertBoxActive = true;
+        this.createStructureApp();
+        this.checkDataForState();
+    }
+
+    checkDataForState() {
+        const getDataFromlocalStorage = localStorage.getItem(this.idOfApp || this.elementInitApp.getAttribute('id'));
+        if(getDataFromlocalStorage) {
+            this.state = JSON.parse(getDataFromlocalStorage);
+            this.renderTodoTasks();
+        } else {
+            this.state = [];
+        }
+    }
+
+    // helper method
+    createElement(elementName, value='', addClass='', id='') {
+        const element =  document.createElement(elementName);
+        if(addClass) element.classList.add(addClass);
+        if(value) element.innerText = value;
+        if(id) element.setAttribute('id', id);
+        return element;
+    }
+
+    createStructureApp() {
+        // create wrapper app
+        const wrapperElementApp = this.createElement('section','','wrapper-todo-app');
+
+        // create Title
+        const h1 = this.createElement('h1');
+        h1.innerText = this.titleOfApp;
+        wrapperElementApp.append(h1);
+
+        const wrapperInsertTask = this.createElement('section','', 'wrapper-insert');
+        const addTaskBtn = this.createElement('button', 'Add Task', 'add-task-btn');
+        const inputElement = this.createElement('input', '', 'input-text-task');
+        inputElement.setAttribute('type','text');
+        inputElement.setAttribute('placeholder','Enter New Task');
+        wrapperInsertTask.append(addTaskBtn);
+        wrapperInsertTask.append(inputElement);
+        wrapperElementApp.append(wrapperInsertTask);
+
+        const wrapperListOfTasks = this.createElement('ul', '', 'wrapper-list-taks');
+        wrapperElementApp.append(wrapperListOfTasks);
+
+        // get context this for use inside function
+        const mainContext = this;
+        // function of event
+        function handlerAddTaskBtn(event) {
+            let value = '';
+            if(this.tagName == 'INPUT') {
```

Comment:
I suggest to use  [Event.target](https://developer.mozilla.org/en-US/docs/Web/API/Event/target) or [Event.currentTarget](https://developer.mozilla.org/en-US/docs/Web/API/Event/currentTarget) rather than the  context `this`

Reply:
thanks for the links, they have been really informative, but yet I do not understand the benefit of them rather than context `this`